tests/copy_observer.cpp

```diff
@@ -50,7 +51,7 @@ void CopyObserver::reset()
 
 CopyObserver &CopyObserver::operator=(CopyObserver &&other) noexcept
 {
-    value = other.value;
+    std::swap(value, other.value);
```

Comment:
leaks `this`'s previous value into `other`

(strictly speaking, not against the rules, but still not nice)